### PR TITLE
Implement control mechanism to check if entered words are correct

### PR DESCRIPTION
**Pull request by bloier**
- Add dictionary API service to get status code (status code "ok" means that word was found in dictionary)
- Set up API key protected as environment variable
- Create new WordController that communicates results to frontend
- Create custom hook useWords which manages the API checks in frontend

### CODE REVIEW

File: backend/src/main/java/de/wordsmith/backend/controller/WordController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@RequestMapping("/api/word")
+public class WordController {
+
+    private final WordflowService wordflowService;
+    private final PonsApiService ponsApiService;
+
+    public WordController(WordflowService wordflowService, PonsApiService ponsApiService) {
+        this.wordflowService = wordflowService;
+        this.ponsApiService = ponsApiService;
+    }
```
Comment (lisamittmann) [inline backend/src/main/java/de/wordsmith/backend/controller/WordController.java:19]: when working with Lombok you can substitute these lines with `@RequiredArgsConstructor` at your class

File: backend/src/main/java/de/wordsmith/backend/controller/WordController.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    @GetMapping()
+    public boolean doesWordExist (@RequestParam String wordText) {
+        return ponsApiService.doesWordExist(wordText);
```
Comment (lisamittmann) [inline backend/src/main/java/de/wordsmith/backend/controller/WordController.java:23]: within domain driven programming using a layer modell I would not expect to see an API class within a controller class, i.e. I would expect you to have an API service class as well as a business logic layer e.g. `ponsWordService`

File: backend/src/main/java/de/wordsmith/backend/service/PonsApiService.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    private final String ponsApi = "https://api.pons.com/v1/dictionary?l=dede&q=";
+    private final RestTemplate restTemplate;
+    @Value("${secret}")
+    private String secret;
+
+    @Autowired
+    public PonsApiService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
```
Comment (lisamittmann) [inline backend/src/main/java/de/wordsmith/backend/service/PonsApiService.java:21]: see comment above, maybe consider `@RequiredArgsConstructor`

File: backend/src/main/java/de/wordsmith/backend/service/PonsApiService.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        try {
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.set("X-Secret", secret);
+            HttpEntity httpEntity = new HttpEntity(httpHeaders);
+            ResponseEntity<Void> response = restTemplate.exchange(requestString, HttpMethod.GET, httpEntity, Void.class);
+            HttpStatus statusCode = response.getStatusCode();
+            return statusCode.equals(HttpStatus.OK);
+        } catch (HttpClientErrorException e) {
+            System.out.println(e.getMessage());
+            return false;
```
Comment (lisamittmann) [inline backend/src/main/java/de/wordsmith/backend/service/PonsApiService.java:34]: simply returning `false` here may be misleading, as the word might still exist but e.g. the API is not responsive at the moment - you may want to consider to pass an error to your frontend allowing you to advise your user to try adding the word at a later time

File: frontend/src/hooks/useWords.js [hunk 5]
```diff
@@ -5,14 +5,20 @@ export default function useWords() {
...
+        .get(`/api/word?wordText=${wordText}`)
+        .then((response) => response.data)
+  }
+
   function addWordToWordsList(wordText) {
     axios
-      .post("/api/letters", { wordText })
+      .post(`/api/word/`, { wordText })
       .then(() => setAllTypedInWords([...allTypedInWords, wordText]))
       .catch((error) => console.error(error.message));
```
Comment (lisamittmann) [inline frontend/src/hooks/useWords.js:18]: you want to consider using an error state which you can use to let your user know that adding the word has not been successful

File: frontend/src/pages/GamePage.js [hunk 6]
```diff
@@ -17,7 +17,12 @@ export default function GamePage() {
 
   const handleSubmit = (event) => {
     event.preventDefault();
-    addWordToWordsList(word);
+    doesWordExist(word)
+    .then((exists) => {
```
Comment (lisamittmann) [inline frontend/src/pages/GamePage.js:21]: `exists` is a misleading wording here, I would expect this value to be a boolean, which you would not add to a list, maybe you want to choose a more appropiate name like `newWord`

**Review by lisamittmann — COMMENTED** [review]
great short PR, just some minor pointers

File: backend/src/main/java/de/wordsmith/backend/controller/WordController.java (hunk 1 above)
Comment (bloier) [reply backend/src/main/java/de/wordsmith/backend/controller/WordController.java:19]: Thanks for the tip! Now implemented in both the controller and service class

File: backend/src/main/java/de/wordsmith/backend/controller/WordController.java (hunk 2 above)
Comment (bloier) [reply backend/src/main/java/de/wordsmith/backend/controller/WordController.java:23]: I do have a WordflowService class, makes sense to first call the method from there. Changed it

File: backend/src/main/java/de/wordsmith/backend/service/PonsApiService.java (hunk 4 above)
Comment (bloier) [reply backend/src/main/java/de/wordsmith/backend/service/PonsApiService.java:34]: Already planned as the next task but thanks for the suggestion :)

**Pull request merged**